extension/lib/display.js

```diff
@@ -0,0 +1,128 @@
+"use strict";
+(function(){
+var namespace = "lib/display.js";
+if (window[namespace] === true) {
+    return;
+} else {
+    window[namespace] = true;
+}
+
+const avg_read_speed = 200; // WPM
+const avg_letters_per_word = 6; // 4.79; just from Quora, add a space after each word
+
+/*
+Creates a display overlaid on current web page that displays
+the estimated time remaining to read and the current speed of 
+the auto-read mode in WPM. 
+*/
+class Display {
+    /*
+    Initialize a display with total time remaining and initial speed (~200 WPM).
+
+    Parameters:
+    - readableDomIds: string[]. List of dom IDs that contain readable content.
+    - speeed: int. Value proportional to the speed of auto-read mode, before adjustment by sentence length. 
+    */
+    constructor(readableDomIds, speed) {
+        this.readableDomIds = readableDomIds; // Used to calc initial reading time
+        this.html = null;
+        this.time_remaining = this.initTimer(readableDomIds);
+        this.reading_speed = this.initSpeed(speed);
+        this.end = null; 
+
+        this.defineHtml();
+        this.createDisplay(readableDomIds);
+    }
+
+    /*
+    Defines html code for display to be inserted into web page. 
+    */
+    defineHtml() {
+        this.html = `
+            <div id="displayContainer">
+                <div id="timerContainer">
+                    <div id="timerInsideContainer">
+                        <span id="timerNumber">${this.time_remaining}</span> min remaining
+                    </div>
+                </div>
+                <div id="speedContainer">
+                    <div id="speedInsideContainer">
+                        <span id="speedNumber">${this.reading_speed}</span> WPM
+                    </div>
+                </div>
+            </div>
+            `;      
+    }
+
+    /*
+    Inserts display HTML into webpage. 
+    */
+    createDisplay(readableDomIds) {
+        document.getElementById(readableDomIds[0]).innerHTML = this.html;
+        document.getElementById("displayContainer").style.opacity = 1; // For smoother transition
+    }
+
+    /* 
+    Params: 
+    - List of container IDs on web page (string[])
+    Return: 
+    The total time remaining to read document, based on avg letters/word and lowball estimate of WPM (int)
+    */
+    initTimer(readableDomIds) {
+        let total_char = 0;
+        let avg_letters_per_word = 6; // Actual is 4.79 (from Quora), add a space after each word to get 6. 
+        let avg_read_speed = 200; // Low ball estimate, from Rayner (in WPM)
+        for (var section in readableDomIds) { // Get total_char of all containers in web page
```

Comment:
I think there should be some sharing of logic w/ the guy who decides how fast autoscrolling should be.
Can you leave a TODO: note for some refactoring of this in a future cl?

Reply:
What do you envision w/ the sharing of logic? Like adjacent/within the display, or tucked away in the extension (requiring user to click on extension to find it)?

Reply:
And yes I actually have the total_char moved out of this function in a different branch, I'll put a TODO comment here and then let me know what you think when I do a pull request for the key word accentuation tomorrow. Also in that version I calculate total # of words which eliminates some of the guesswork (instead of going from total number of characters -> words by using the average number of letters per word)

Reply:
There should be one definition of this logic:
- given [text string, current speed setting], return [estimated time]

Then, this logic will be used in two places:
- To figure out how long auto-scroll should linger in current highlighted sentence
- For the display to estimate how long reading all the sentences will take.

Which makes me think...
Might consider precalculating the reading time of everything (each [containerId, offset] pair) and storing it somewhere. Then, the two use cases (auto-scroll, reading time estimate) can just rely on the precalculated reading time.

And yeah, definitely not in this CL, will see how the code looks after ur other pull